src/pages/transactions/_components/DfTx/DfTxCompositeSwap.tsx

```diff
@@ -12,19 +12,50 @@ import { TokenSymbol } from "@components/commons/token/TokenSymbol";
 import { PoolPairSymbol } from "@components/commons/token/PoolPairSymbol";
 import { IoArrowForwardOutline } from "react-icons/io5";
 import NumberFormat from "react-number-format";
+import { Transaction } from "@defichain/whale-api-client/dist/api/transactions";
+import { useEffect, useState } from "react";
+import { useWhaleApiClient } from "@contexts/WhaleContext";
+import BigNumber from "bignumber.js";
 import { DfTxHeader } from "./DfTxHeader";
 
 interface DftxCompositeSwapProps {
   dftx: DfTx<CompositeSwap>;
+  transaction?: Transaction;
 }
 
 export function DfTxCompositeSwap({
   dftx: { data },
+  transaction,
 }: DftxCompositeSwapProps): JSX.Element {
   const network = useNetwork().name;
-
   const toAddress = fromScript(data.poolSwap.toScript, network);
   const fromAddress = fromScript(data.poolSwap.fromScript, network);
+  const [toAmount, setToAmount] = useState<string>();
+  const api = useWhaleApiClient();
+
+  useEffect(() => {
+    void getAccountHistory();
+  }, [transaction]);
+
+  async function getAccountHistory() {
```

Comment:
can we rename this to `getSwapToAmount` and then return `toAmount`, then we `setToAmount(getSwapToAmount())` at line 37?